static/js/services/alertify.js

```diff
@@ -104,6 +113,57 @@ define(["angular"], function(angular) {
 				}
 				return dlg;
 			},
+			/*
+			* @description
+			* Create custom dialog with custom template. Currently the following custom dialog
+			* ids are publicly overwrittable:
+			* - webrtcUnsupported: Dialog to inform the user that their browser is not supported
+			*
+			* If no config Object is provided then the existing config object for that
+			* id will be retrieved and used.
+			*
+			* To completely overwrite the template of an existing custom dialog overwrite the
+			* $templateCache url entry.
+			*
+			* @param {string} id The ID of the custom dialog.
+			* @param {Object} config Preferences for the dialog with the following properties:
+			* <ul>
+			*   <li>**baseType**: {string} existing dialog type to use as initial values.</li>
+			*   <li>**type**: {string} the template name which is saved in the $templateCache at templateUrl. If the templateUrl is '/dialogs/notify.html' the type would be notify</li>
+			*   <li>**title**: {string} the title which the modal dialog should display.</li>
+			*   <li>**message**: {string} the message which the modal dialog should display.</li>
+			*   <li>**ok_cb**: {Function} the callback function to be called on success.</li>
+			*   <li>**err_cb**: {Function} the callback function to be called on error.</li>
+			* </ul>
+			*/
+			custom: function(id, config) {
+				var retrieval = false;
+				if (typeof config !== 'object') {
```

Comment:
All this config stuff should go into a registry which can be called. It should not be possible to load a custom dialog which was not registered before.